src/ibex_bluesky_core/callbacks/file_logger.py

```diff
@@ -79,9 +79,11 @@ def start(self, doc: RunStart) -> None:
         )
         header_data[START_TIME] = formatted_time
 
+        if not self.filename.parent.exists():
+            self.filename.parent.mkdir()
```

Comment:
Usually ` os.makedirs(name, exist_ok=True)` is recommended as:
- It makes intermediate directories
- It's less likely to suffer from race conditions between the check for dir existing and creation